### DISCUSSION TITLE
Pointers for I/O integration with WinForms application

### DISCUSSION THREAD

**Discussion by bcalco** [leppie/IronScheme#84] [Discussions]
@leppie 

About to work on some more deliberate interaction between the GUI and IronScheme engine. 

* Custom REPL/Console for interactive development (both command line and GUI based)
* Hooking input/output (for example, if a user calls `(display...)` or includes a user prompt from the GUI environment (like a generic input box I can create)

Before I embark on these things I thought it wise to ask for some pointers from you, and anyone else who may have done such things already, so I can plan what I need to do in what order etc.

I also expect to extend the "core library" with Scheme code, to which my users will have access, that needs access to the internals of my application, both in the main EXE and in several satellite libraries I'm creating. I'm not finding the wiki entries fully expositive on how to do this in a way I can dev locally, and build for deployment to customer machines. Any help or examples/sample apps along these lines would be appreciated.

**Comment by leppie** [comment]
IronScheme uses standard .NET input/output ports/streams. 

Here is some pointers from code that used to be in the repo, but removed when I added .NET Core support.

https://github.com/IronScheme/IronScheme/blob/16ebd2f505cc5b431d805e86200db37dda550675/IronScheme/IronScheme.Web.Eval/web-eval.sls

**Comment by leppie** [comment]
Also here again
https://github.com/IronScheme/IronScheme.VisualStudio2/blob/master/IronScheme.VisualStudio.REPL/CommandBuffer.cs

**Comment by bcalco** [comment]
@leppie - Thanks for your feedback, I'll have lots more questions, and now that I see you've moved this to discussions, I'll try to post such questions here instead of on the Issues tabs. That will be for real bugs or enhancement requests.
